tests/rbd_mirror/test_configure_failback_mirror_snapshot.py

```diff
@@ -0,0 +1,48 @@
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """
+    Creates user specified Pool and enables snapshot based Mirroring
+    on each images in that pool.
+    Args:
+        **kw: test data
+        Example::
+            config:
+                imagesize: 2G
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+    try:
+        log.info("Configuring RBD mirroring with snapshot based")
+        config = kw.get("config")
+        number_of_images = config.get("image-count", 20)
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        # creating user specified images to verify images getting replicated after mirror failback
+        for i in range(1, number_of_images):
+            imagename = "tier_2_rbd_mirror_image" + mirror1.random_string()
+            mirror1.initial_mirror_config(
+                mirror2,
+                poolname=config.get("poolname"),
+                imagename=imagename,
+                imagesize=config.get("imagesize", "1G"),
+                io_total=config.get("io-total", "1G"),
+                mode="image",
+                mirrormode="snapshot",
+                **kw,
+            )
```

Comment:
initial_mirror_config tried to create pool and tried to bootstrap mirroring between the cluster
IMO we should avoid redundant execution of the steps!

We are executing many extra commands on each cluster number_of_images times, I hope we work on avoiding this.

Reply:
let's discuss the solutions